src/ripple/shamap/impl/SHAMapSync.cpp

```diff
@@ -419,7 +419,7 @@ SHAMapAddNode SHAMap::addRootNode (Blob const& rootNode,
 
     assert (seq_ >= 1);
     auto node = SHAMapAbstractNode::make(
-        rootNode, 0, format, uZero, false, f_.journal ());
+        rootNode, 0, format, SHAMapHash{uZero}, false, f_.journal ());
```

Comment:
If on line 29 you declare `uZero` as `static const SHAMapHash uZero {beast:zero}`, then I think you can get rid of this local `SHAMapHash` temporary and two others further down in the file.